File: Competicoes Diversas/2020/Rinha de Calouro/Pacotes/D/solver/D.py
import math

a,b =[int(x) for x in input().split()]
mmc=4
v=[int(x) for x in input().split()]
for i in range(a):
    mmc=(mmc*v[i])//(math.gcd(mmc,v[i]))
res = 2020+mmc
minimo = 2020+b

while(res<minimo):
    res+=mmc
print(res)

File: Competicoes Diversas/2020/Rinha de Calouro/Pacotes/D/solver/D.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void){
    int n,d;
    cin >> n >> d;
    int mmc=4;
    for(int i=0;i<n;i++){
        int a;
        scanf("%d", &a);
        mmc=(mmc*a)/__gcd(mmc,a);
    }
    int resposta = 2020+mmc;
    int minimo = 2020+d;
    while(resposta<minimo){
        resposta+=mmc;
    }
    printf("%d\n", resposta);
}